pkg/plugins/resources/bitbucket/client/main.go

```diff
@@ -0,0 +1,96 @@
+package client
+
+import (
+	"encoding/base64"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/drone/go-scm/scm"
+	"github.com/drone/go-scm/scm/driver/stash"
+	"github.com/drone/go-scm/scm/transport/oauth2"
+	"github.com/sirupsen/logrus"
+)
+
+// Spec defines a specification for a "bitbucket" resource
+// parsed from an updatecli manifest file
+type Spec struct {
+	// [S][C][T] URL specifies the default github url in case of Bitbucket enterprise
+	URL string `yaml:",omitempty" jsonschema:"required"`
+	// [S][C][T] Username specifies the username used to authenticate with Bitbucket API
+	Username string `yaml:",omitempty"`
+	// [S][C][T] Token specifies the credential used to authenticate with
+	Token string `yaml:",omitempty"`
+	// [S][C][T] Token specifies the credential used to authenticate with
+	Password string `yaml:",omitempty"`
+	// Owner specifies repository owner
+	Owner string `yaml:",omitempty" jsonschema:"required"`
+	// Repository specifies the name of a repository for a specific owner
+	Repository string `yaml:",omitempty" jsonschema:"required"`
+}
+
+type Client *scm.Client
+type BasicAuthTransport struct {
+	Username string
+	Password string
+}
+
+func (bat BasicAuthTransport) RoundTrip(req *http.Request) (*http.Response, error) {
```

Comment:
```suggestion
func (bat BasicAuthTransport) RoundTrip(req *http.Request) (*http.Response, error) {
```

In Go, any function starting with Upper case are public function that can be used by third Golang package. 
While I don't consider this as blocker since I have a lot of left over, I now try to have all function be private by using lower case by default unless of a good reason